server/validate/email/validate.go

```diff
@@ -119,6 +121,31 @@ func isTemplateValid(templ *textt.Template) error {
 	}
 	return nil
 }
+type SMPTLogin struct {
+	username, password string
+}
+
+func LoginAuth(username, password string) smtp.Auth {
+	return &SMPTLogin{username, password}
+}
+
+func (a *SMPTLogin) Start(server *smtp.ServerInfo) (string, []byte, error) {
+	return "LOGIN", []byte(a.username), nil
+}
+
+func (a *SMPTLogin) Next(fromServer []byte, more bool) ([]byte, error) {
+	if more {
+		switch string(fromServer) {
+		case "Username:":
+			return []byte(a.username), nil
+		case "Password:":
+			return []byte(a.password), nil
+		default:
+			return nil, errors.New("Unknown from server")
```

Comment:
Maybe make the error more descriptive, like 'Unknown server response in LOGIN AUTH' or something like that.